assume/strategies/flexable_storage.py

```diff
@@ -130,7 +130,8 @@ def calculate_bids(
 
             # if price is higher than average price, discharge
             # if price is lower than average price, charge
-            if price_forecast[start] >= average_price:
+            # if price forecast favors discharge, but max discharge is zero, set a bid for charging
+            if price_forecast[start] >= average_price and max_power_discharge[start]:
```

Comment:
Why not do nothing in this case?

Reply:
Because we would waste a bidding possibility without any benefit.

As we can only bid either positive or negative, we always try to bid on what is more likely (by looking at the mean).
But if we can not bid the most likely, due to our storage constraints, it is best to bid the other direction instead of not doing anything?

Reply:
But this would enforce the following behaviour in some cases:
We have a very high price of 300 €/MWh, and yet we cannot discharge due to technical constraints. We will bid for charging for 300 €/MWh.
Since we also place a price bid at the level of average_price...

I would opt for doing nothing in these cases.

Reply:
Given our average is 150€/MWh and the efficiency is 0.8
We expect a price which is higher than average (300€/MWh).
We would like to discharge (and set a price of 150€/0.8 = 187.5€), but we can't.

So we should instead set a bid for charging, but do respect the average and efficiency as well:
We set 150*0.8=120€/MWh.

This is fine and not harmful, because we still rely on the average price and do not set `price_forecast[start] / efficiency` (I think you expected it that way)